### PR TITLE
Updated the Project form for the new `preset_questions` field

### LINKED ISSUE #84: New chat page title, description and preset questions should be fetched from the project endpoint
needs [#](https://github.com/coronasafe/ayushma/issues/263) to be merged

image

### PR DESCRIPTION
**Pull request by GokulramGHV**
fixes #84 

Waiting for coronasafe/ayushma#263 to be merged

image

### CODE REVIEW

**Review by Ashesh3 — APPROVED** [review]

File: src/app/(main)/project/[project_id]/page.tsx [hunk 1]
```diff
@@ -108,12 +118,18 @@ export default function Chat(params: { params: { project_id: string } }) {
...
+  const samplePrompts = useMemo(
+    () =>
+      project?.preset_questions && project?.preset_questions?.length > 0
+        ? project?.preset_questions?.sort(() => Math.random() - 0.5).slice(0, 4)
+        : [
+            "How does one prepare for advanced airway management, even before the patient has arrived?",
+            "How do you assess a ICU patient's need for tracheal intubation?",
+            "What are the checks one needs to perform before intubation?",
+            "How do you do the initial assessment of the patient before oxygenation or ventilation?",
+          ],
```
Comment (bodhish) [inline src/app/(main)/project/[project_id]/page.tsx:130]: Why should the app assume that the questions will always be about `Septic Shock`?

One of the bug that I spotted was the UI was showing this questions till the preset_questions are loaded

File: src/app/(main)/project/[project_id]/page.tsx [hunk 2]
```diff
@@ -108,12 +118,18 @@ export default function Chat(params: { params: { project_id: string } }) {
...
-  const samplePrompts = [
-    "How does one prepare for advanced airway management, even before the patient has arrived?",
-    "How do you assess a ICU patient's need for tracheal intubation?",
-    "What are the checks one needs to perform before intubation?",
-    "How do you do the initial assessment of the patient before oxygenation or ventilation?",
-  ];
+  const samplePrompts = useMemo(
+    () =>
+      project?.preset_questions && project?.preset_questions?.length > 0
+        ? project?.preset_questions?.sort(() => Math.random() - 0.5).slice(0, 4)
```
Comment (bodhish) [inline src/app/(main)/project/[project_id]/page.tsx:124]: Why are you sorting the questions here? Can't the server respond with random questions in each request;

**Review by bodhish — COMMENTED** [review]
Check this out @Ashesh3

File: src/app/(main)/project/[project_id]/page.tsx (hunk 1 above)
Comment (Ashesh3) [reply src/app/(main)/project/[project_id]/page.tsx:130]: @bodhish Do you want to not show any prompts if no custom prompts are defined or would you prefer to have a configurable global list of default prompts to show if a project has no custom prompts.

File: src/app/(main)/project/[project_id]/page.tsx (hunk 2 above)
Comment (Ashesh3) [reply src/app/(main)/project/[project_id]/page.tsx:124]: That is possible, though I do not see the need to offload this extra logic to backend, the random pick n choose is fine to do on the client side, right? Unless the full prompt list is supposed to be private (which doesn't seem to be the case)..

File: src/app/(main)/project/[project_id]/page.tsx (hunk 1 above)
Comment (bodhish) [reply src/app/(main)/project/[project_id]/page.tsx:130]: Don't show anything unless the questions are set in the project

File: src/app/(main)/project/[project_id]/page.tsx (hunk 2 above)
Comment (bodhish) [reply src/app/(main)/project/[project_id]/page.tsx:124]: Say I have 100 questions, why should the FE load 100 when its just rendering 4; 🤔

Comment (Ashesh3) [reply src/app/(main)/project/[project_id]/page.tsx:124]: Makes sense, I didn't consider that case. 😅

File: src/app/(main)/project/[project_id]/page.tsx (hunk 1 above)
Comment (Ashesh3) [reply src/app/(main)/project/[project_id]/page.tsx:130]: Resolved in https://github.com/coronasafe/ayushma_fe/pull/100

**Pull request merged**